### PR TITLE
4.5.5 - Remove Jenkins and replace with GitHub Actions

### PR DESCRIPTION
**Pull request by Wolftousen**
As the title says.  It also enhances the exercises a bit and links them up with the upcoming GitHub chapter

### CODE REVIEW

File: docs/4-software-development-practices/4.5.5-test-automation.md [hunk 1]
```diff
@@ -4,36 +4,40 @@ docs/4-software-development-practices/4.5.5-test-automation.md:
...
+Automated testing is a crucial step to ensure code quality because developed tests are guaranteed to run once new source code is committed. GitHub Actions is a tool that can help us leverage our tests in a build step. With a testing stage in GitHub Actions, code quality can be ensured by automatically running the tests that have been written. If tests fail, the build can be marked as unstable and deployment will not occur.
 
-## Exercise
+## Exercise 1
 
-1. Locate the Jenkins Dockerfile created in the container section of the bootcamp.
-2. Edit this Dockerfile to add a Go installation to the image.
-3. Create a Freestyle Project that will point to your completed Go Unit Testing exercise from your fork of the Bootcamp.
+1. Create a new action, `test`, like you did for the `build` action in the `GitHub Actions` section of the bootcamp.
```
Comment (jburns24) [inline docs/4-software-development-practices/4.5.5-test-automation.md:21]: I would remove reference to the `GitHub Actions` section of the bootcamp as I am not sure when that initiative will get finished or added. The bootcamp as it is is a rather gross offender of saying "do x" without introduction to "x". I am ok if we do that in this situation

File: docs/4-software-development-practices/4.5.5-test-automation.md [hunk 2]
```diff
@@ -4,36 +4,40 @@ docs/4-software-development-practices/4.5.5-test-automation.md:
...
+    * Setup - Setup the Go Language like you did in the `build` action
+    * Test - Run your Unit Tests
 
-?>Note: After pointing to your forked repo in Jenkins, you will need to get the path to the Go exercises in the build step
+## Exercise 2
 
-4. Add a build step for testing your exercise.
-5. Add a build step to build the exercise.
-6. Build the Jenkins job and confirm the job has run the tests and the build was successful.
+1. Open up your `build` action that you created in the `GitHub Actions` section of the bootcamp.
```
Comment (jburns24) [inline docs/4-software-development-practices/4.5.5-test-automation.md:29]: Assuming this wording is based on the assumption that the GitHub Actions new section will be finished and merged before you section. If so let's not assume that. 

If it is not based on that assumption I must have a missunderstanding

File: .cspell.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
         "AWSCLIV",
         "azurerm",
         "blaa",
+        "bootcamp",
```
Comment (jburns24) [inline .cspell.json:13]:
```suggestion
```

NP: Should not actually be required with the bug in this file fixed

**Review by jburns24 — APPROVED** [review]

**Comment by jburns24** [comment]
Want to hold off on merging this until @rhoofard wraps up the GitHub chapter

**Comment by amber-beasley-liatrio** [comment]
@Wolftousen , I saw you wanted to wait until the GitHub chapters were done before merging your changes and I saw you had references to the GitHub chapter.

I also wanted to piggy back off of the GitHub chapter so I can reference any sample/getting started project a camper has already created.

There is a bit of a continuity problem. The chapter we are working on are in 4 but GitHub is in Chapter 9 

image

So they would read our chapters first before getting to the GitHub chapter. 

@rhoofard (maybe we can sync up about order of the chapters?)

**Comment by jburns24** [comment]
@amber-beasley-liatrio When I touched bases with Ryan this morning I mentioned that I wanted this moved to Chapter 2. I think he has been swamped with BlueYonder work and hasnt touched this in few days. But he is aware that it should be moved to chapter 2.

**Pull request merged**